great_expectations/expectations/metrics/column_aggregate_metrics/column_mean.py

```diff
@@ -32,6 +32,11 @@ def _sqlalchemy(cls, column, **kwargs):
     def _spark(cls, column, _table, _column_name, **kwargs):
         """Spark Mean Implementation"""
         types = dict(_table.dtypes)
-        if types[_column_name] not in ("int", "float", "double", "bigint"):
-            raise TypeError("Expected numeric column type for function mean()")
+        if (
+            types[_column_name] not in ("int", "float", "double", "bigint")
+            and "decimal" not in types[_column_name]
```

Comment:
@TrangPham Would it work if we added `"decimal"` to the `tuple` above?  Thank you.

Reply:
No as it actually returns “decimal(x, y)” where x and y are ints. I believe they are for scale and precision.

Reply:
but then `and "decimal" not in types[_column_name]` -- seems to do the same?..

Reply:
I think I see what is going on; seems like we should handle this more robustly than just string inclusion (if possible).

Reply:
@TrangPham I am thinking that if we use `_table.schema[_column_name].dataType`, then we will get formal `PySpark` type information (`StringType`, `IntegerType`, `DecimalType`, etc.), and then we can do the checks for which of them satisfies the "numeric type" requirements.  Please note that if you take this approach, you would need to modify the `Spark` compatibility import to also include `types` and be able to reference `types.StringType`, `types.IntegerType`, `types.DecimalType`, etc. as needed.  Thank you!

Reply:
Will update! Thanks for finding a more robust solution!